servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/PrioritizedHost.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+/**
+ * Handles for determining the weight of a host based on it's underlying weight and priority.
+ */
+public interface PrioritizedHost {
+
+    /**
+     * The current priority of the host.
+     * @return the current priority of the host.
+     */
+    int priority();
+
+    /**
+     * Whether the host is considered healthy or not.
+     * @return whether the host is considered healthy or not.
+     */
+    boolean isHealthy();
+
+    /**
+     * The intrinsic weight of the host.
+     * @return the intrinsic weight of the host.
+     */
+    double weight();
+
+    /**
+     * Set the weight of the host to use during load balancing.
```

Comment:
Apologies if this is obvious but I am a little bit confused about the description of weight. Is there a difference between "intrinsic weight" and "weight during load balancing" -- and if the setter adjusts the weight does it also change the weight of the getter?

Reply:
Good eye! I had names that were slightly different earlier. I understand the confusion. I changed the name to `loadBalancingWeight` to be more clear.